extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {closest} from '../../../src/dom';
+import {dev, user} from '../../../src/log';
+import {isExperimentOn} from '../../../src/experiments';
+
+export class AmpSelector extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {boolean} */
+    this.isMultiple_ = false;
+
+    /** @private {!Array<!Element>} */
+    this.selectedOptions_ = [];
+
+    /** @private {!Array<!Element>} */
+    this.options_ = [];
+
+    /** @private {!Array<!Element>} */
+    this.inputs_ = [];
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    return true;
+  }
+
+  /** @override */
+  buildCallback() {
+    user().assert(isExperimentOn(this.win, 'amp-selector'),
+        `Experiment amp-selector is disabled.`);
+    this.isMultiple_ = this.element.hasAttribute('multiple');
+    const isDisabled = this.element.hasAttribute('disabled');
+
+    this.element.setAttribute('role', 'listbox');
+
+    if (this.isMultiple_) {
+      this.element.setAttribute('aria-multiselectable', 'true');
+    }
+
+    if (isDisabled) {
+      this.element.setAttribute('aria-disabled', 'true');
+    }
+
+    if (!isDisabled) {
+      this.init_();
+      this.element.addEventListener('click', this.clickHandler_.bind(this));
+    }
+  }
+
+  /**
+   * @private
+   */
+  init_() {
+    const options = [].slice.call(this.element.querySelectorAll('[option]'));
+    options.forEach(option => {
+      option.setAttribute('role', 'option');
+      if (option.hasAttribute('disabled')) {
+        option.setAttribute('aria-disabled', 'true');
+        option.removeAttribute('selected');
```

Comment:
I tested with normal forms - anything that is disabled doesnot get submitted regardless. So we can still have the selected but it would do nothing.